Updates for Katello 4.10.0 - Release notes (#2481)
theforeman/foreman-documentation: guides/doc-Release_Notes/topics/katello.adoc

[id="katello-release-notes"]
= Katello {KatelloVersion} Release Notes

[id="katello-upgrade-warnings"]
== Upgrade Warnings

=== Katello-Agent has been removed.
Katello-Agent has been removed from Katello and is no longer available via web UI, API, or CLI.
katello-agent will be removed by upgrading, REX alternative will not be installed as part of the upgrade.

[id="katello-headline-features"]
== Headline Features

> === Pulpcore and plugins upgraded to Pulpcore 3.28.

Reviewer: We always write a bit of text under a header, explaining why it is a headline feature and why they want it.

[id="katello-headline-features"]
== Headline Features

=== Pulpcore and plugins upgraded to Pulpcore 3.28.

> Repositories now use pulp_deb 3.0.0.

Reviewer: I'd explain this a bit further. For example, Katello 4.9 used Pulp 3.22 so you can mention that, as well as some of the major features since Pulp 3.22.

=== Hammer: Allow user to run Verify Content Checksum, on container repositories.

=== Add ACS refresh option '--all' to "hammer alternate-content-source refresh" command, to refresh all ACS\'s.
ACS bulk refresh through API silently sanitizes input ids.

=== Remove the setting \'Use remote execution by default\'.
As part of the removal of katello-agent, the following related settings have been removed: 
\'Use remote execution by default\', \'Accept action timeout\', and \'Finish action timeout.\'

=== Add activation key details top bar.
When the setting 'Show Experimental Labs' is enabled, 
the Activation Key details page will be replaced by the beginnings of a redesigned page.

=== Content Views and CVV enhancements to filters and refresh.

[id="katello-upgrade-warnings"]
== Upgrade Warnings

> === Katello-Agent has been removed.

Reviewer: ```suggestion
=== Katello-Agent removal
```

[id="katello-headline-features"]
== Headline Features

=== Verify Content Checksum on container repositories using Hammer
Option '--validate-contents' now supports all repository types, using "hammer repository synchronize" command. 
Previously container repositories and some others were not supported.

=== Add ACS bulk refresh option
Added option '--all' to "hammer alternate-content-source refresh" command.

=== Experimental activation key details page
When the setting 'Show Experimental Labs' is enabled, the Activation Key details page will be replaced by the beginnings of a redesigned page.

=== Content Views and CVV enhancements to filters and refresh
Content View filter creation/edit UI: 'Edit Rule' button is now 'Save'.
If dependency solving is enabled or disabled for a content view, it will now be marked as needing a new publish.

=== Pulpcore and plugins updated from Pulpcore 3.22 to 3.28

> Contains both fixes and features that Katello can use in the future to better manage content, such as:

Reviewer: I think you need an empty line in between:
```suggestion
Contains both fixes and features that Katello can use in the future to better manage content, such as:

```

Author: In latest push

[id="katello-headline-features"]
== Headline Features

> === Verify Content Checksum on container repositories using Hammer

Reviewer 2: ```suggestion
=== Verify Content Checksum on container repositories using Hammer

```

Added option '--validate-contents true/false' to "hammer repository synchronize" command.

=== Add ACS bulk refresh option
Added option '--all' to "hammer alternate-content-source refresh" command.

> ACS bulk refresh through API silently sanitizes input ids.

Reviewer 3: ```suggestion

```

I don't think this is needed, just what's on line 11 should be enough. What's on line 12 is a bug that occurred during development of 4.10.

> === Experimental activation key details page

Reviewer 2: ```suggestion
=== Experimental activation key details page

```

When the setting 'Show Experimental Labs' is enabled, the Activation Key 
details page will be replaced by the beginnings of a redesigned page.

=== Content Views and CVV enhancements to filters and refresh
Content View filter creation/edit button now says 'Save' instead of 'Edit Rule'.

> Content view dependency solving tracked for needs_publish.

Reviewer 3: ```suggestion
If dependency solving is enabled or disabled for a content view, it will now be marked as needing a new publish.
```

> === Pulpcore and plugins updated from Pulpcore 3.22 to 3.28

Reviewer 2: ```suggestion
=== Pulpcore and plugins updated from Pulpcore 3.22 to 3.28

```

Contains both fixes and features that Katello can use in the future to better manage content, such as:

* https://github.com/pulp/pulp_rpm/issues/3135[More filter options on the packages API]  
* https://github.com/pulp/pulp_rpm/issues/3009[New setting for generated primary.xml timestamps to match upstream repository]  
* https://github.com/pulp/pulp_rpm/issues/2995[RPM replicator support]  
* https://github.com/pulp/pulpcore/issues/3538[New distribution hiding option]  
* https://github.com/pulp/pulpcore/issues/3835[OpenTelemetry support]  
* https://github.com/pulp/pulpcore/issues/3656[File size added to content application listing]  

[id="katello-upgrade-warnings"]
== Upgrade Warnings

> === Katello-Agent removal

Reviewer 2: ```suggestion
=== Katello-Agent removal

```

Katello-Agent has been removed from Katello and is no longer available via web UI, API, or CLI.

> katello-agent will be removed by upgrading.

Reviewer 3: ```suggestion
Katello-Agent will be removed by upgrading.
```

For patching hosts going forward, use remote execution.
Remote execution will not be automatically enabled as part of the upgrade. 
To ensure uninterrupted host patching, migrate to remote execution before upgrading.
To migrate hosts, see {ManagingHostsDocURL}Migrating_Hosts_From_Katello_Agent_to_Remote_Execution_managing-hosts[Migrating from Katello Agent to Remote Execution] in _{ManagingHostsDocTitle}_.

> === Removed various settings

Reviewer 2: ```suggestion
=== Removed various settings

```

Author: Thanks I will get these pushed soon!

[id="katello-headline-features"]
== Headline Features

> === Allow user to run Verify Content Checksum using Hammer, on container repositories

Reviewer: I'd keep it even shorter
```suggestion
=== Verify Content Checksum on container repositories using Hammer
```

> Added option '--validate-contents true/false' to "hammer repository synchronize" command.

Reviewer: Please take this as a suggestion on how to write headline features. I'm taking this one because it's the first one, but you can ask yourself the same questions for other headline features.

When I read this line I have more questions than answers. For example, what was the previous behavior? Was it never validating or always? What is the effect if I pass one or the other? Why would I choose to not verify checksums, because I'd think that's always a good thing.

The preview is rendered. Please compare the Foreman headline features (https://theforeman-foreman-documentation-preview-pr-2481.surge.sh/3.8/Release_Notes/index-katello.html#foreman-headline-features) with the Katello ones. With those I've tried to compare the previous state to the new state and explain why the new state is better. I won't claim to be very good at it, but that's at least the thought process behind it.

Reviewer 3: Yeah @damoore044 this one is bit confusing -- context is that we've always had this `--validate-contents` flag, but it didn't work for container repositories (and some others) due to some historical reason. So, we just "unlocked" `--validate-contents` for all repo types.  It was all a backend change, not even a change in hammer.

Also, `--validate-contents` actually triggers a different task in Pulp that's not technically even a repository sync. Back in Pulp 2, a validate content sync would also sync the repository (if I remember right). But, in Pulp 3, validating contents is a specific task that just checks for proper checksums.

> Option '--validate-contents' now supports all repository types,
> using "hammer repository synchronize" command. 

Reviewer 4: ```suggestion
Option '--validate-contents' now supports all repository types, using "hammer repository synchronize" command. 
```

> === Add ACS bulk refresh option, to refresh all ACS's

Reviewer: ```suggestion
=== Add ACS bulk refresh option
```

Added option '--all' to "hammer alternate-content-source refresh" command.
ACS bulk refresh through API silently sanitizes input ids.

> === Add activation key details top bar

Reviewer: To me it wasn't clear what it was. How about:
```suggestion
=== Experimental activation key details page
```

=== Experimental activation key details page

> When the setting 'Show Experimental Labs' is enabled, the Activation Key 
> details page will be replaced by the beginnings of a redesigned page.

Reviewer 4: ```suggestion
When the setting 'Show Experimental Labs' is enabled, the Activation Key details page will be replaced by the beginnings of a redesigned page.
```

=== Content Views and CVV enhancements to filters and refresh
Content View filter creation/edit button now says 'Save' instead of 'Edit Rule'.
Content view dependency solving tracked for needs_publish.

=== Pulpcore and plugins updated from Pulpcore 3.22 to 3.28
Contains both fixes and features that Katello can use in the future to better manage content, such as:
https://github.com/pulp/pulp_rpm/issues/3135[More filter options on the packages API]
https://github.com/pulp/pulp_rpm/issues/3009[New setting for generated primary.xml timestamps to match upstream repository]
https://github.com/pulp/pulp_rpm/issues/2995[RPM replicator support]
https://github.com/pulp/pulpcore/issues/3538[New distribution hiding option]
https://github.com/pulp/pulpcore/issues/3835[OpenTelemetry support]

> https://github.com/pulp/pulpcore/issues/3656[File size added to content application listing]

Reviewer: This now renders as one long line. Did you intend to make it a list?

Reviewer 3: Yeah it should be a list... does it need newlines in between them?

Author: Is `\` the proper newline delimiter? 
Do we have a new rendered preview @ekohl ?

[id="katello-upgrade-warnings"]
== Upgrade Warnings

=== Katello-Agent removal
Katello-Agent has been removed from Katello and is no longer available via web UI, API, or CLI.
katello-agent will be removed by upgrading.
For patching hosts going forward, use remote execution.
Remote execution will not be automatically enabled as part of the upgrade. 
To ensure uninterrupted host patching, migrate to remote execution before upgrading.
Guide for migrating to remote execution:
[https://github.com/theforeman/foreman-documentation/blob/3.8/guides/common/modules/proc_migrating-hosts-from-katello-agent-to-remote-execution.adoc].

> === Remove the setting 'Use remote execution by default'

Reviewer: The heading says one setting is removed, but the body mentions 3 settings.

As for a suggestion I'm debating on a simple heading like:
```suggestion
=== Removed various settings
```

Or since it's related to the agent removal you can also squash it into the previous section.

=== Pulpcore and plugins upgraded to Pulpcore 3.28
Repositories now use pulp_deb 3.0.0.
Katello 4.9 used Puplcore 3.22, some features since 3.23.
GCP support [3.23].
Added a repository exact and in filters to the Distributions API [3.23].
Added ability to filter resources by list of HREFs or IDs with pulp_href__in and pulp_id__in [3.24].
Added commands and documentation for db encryption key rotation [3.24].
Added filters with_content and latest_with_content to Repository list endpoint [3.25].
Updated Django version to 4.2 LTS branch [3.25].
Enabled OpenTelemetry metrics and tracing for the pulp-api app [3.26].
Expose user who dispatched a Task under new field created_by [3.27].
Added hidden field to Distributions [3.27].
Added support to pull-through caching for streaming metadata files [3.28].
Added support to pull-through caching for plugins with multi-artifact content types [3.28].
Added Distribution.content_headers_for() to let plugins affect content-app response headers [3.28].

[id="katello-upgrade-warnings"]
== Upgrade Warnings

=== Katello-Agent removal
Katello-Agent has been removed from Katello and is no longer available via web UI, API, or CLI.

> katello-agent will be removed by upgrading, REX alternative will not be installed as part of the upgrade.

Reviewer 5: ```suggestion
For patching hosts going forward, use remote execution.
katello-agent will be removed by upgrading.
Migrate to remote execution before upgrading, as REX will not be automatically enabled as part of the upgrade.
```

Reviewer: I'd say you don't need to enable REX before updating. REX is always installed with Katello, but it's not on every content proxy. You can do that after you upgrade though.

You also need to make sure the hosts are updated though. I'm not sure you can do that via katello-agent today. If you can, that's a reason to do it before upgrading Katello. Otherwise I don't see a difference.

Reviewer 5: It's true that it won't break anything if you wait to enable REX until after upgrading. But on the other hand, you won't be able to patch your hosts until you enable REX. So "Migrate before upgrading" is more of a recommendation.

Reviewer 5: Maybe this?

```
Remote execution will not be automatically enabled as part of the upgrade. To ensure uninterrupted host patching, migrate to remote execution before upgrading.
```

Reviewer: We have a guide on that as well. It'd be good to link to it.

And a minor preference to do it after upgrading is that it's trickier to link between versions, while we have infrastructure in place to link within the same version. You could argue it's within the same upgrade window, but technically it would be correct that there's less downtime in ability to patch if you perform it on the previous version.

Author: @ekohl @jeremylenz do you have a link to the latest guide on migrating to REX?

Reviewer 5: It's here: https://docs.theforeman.org/nightly/Managing_Hosts/index-katello.html#Migrating_Hosts_From_Katello_Agent_to_Remote_Execution_managing-hosts

but I'm not sure of the correct way to link it via ASCIIdoc.

Also, if the docs were properly updated for Katello 4.10, that section wouldn't even be there, so I'm not quite sure what to do here.

Author: https://github.com/theforeman/foreman-documentation/blob/3.8/guides/common/modules/proc_migrating-hosts-from-katello-agent-to-remote-execution.adoc

[id="katello-headline-features"]
== Headline Features

=== Allow user to run Verify Content Checksum using Hammer, on container repositories
Added option '--validate-contents true/false' to "hammer repository synchronize" command.

=== Add ACS bulk refresh option, to refresh all ACS's
Added option '--all' to "hammer alternate-content-source refresh" command.
ACS bulk refresh through API silently sanitizes input ids.

=== Add activation key details top bar
When the setting 'Show Experimental Labs' is enabled, the Activation Key 
details page will be replaced by the beginnings of a redesigned page.

=== Content Views and CVV enhancements to filters and refresh
Content View filter creation/edit button now says 'Save' instead of 'Edit Rule'.
Content view dependency solving tracked for needs_publish.

> === Pulpcore and plugins upgraded to Pulpcore 3.28

Reviewer 3: I have a suggestion for some features to pull out that might interest users. Many of these features are things that Katello hasn't integrated with.

So, I would replace this whole section with something like:

```
=== Pulpcore and plugins updated to Pulpcore 3.28
Contains both fixes and features that Katello can use in the future to better manage content, such as:
https://github.com/pulp/pulp_rpm/issues/3135[More filter options on the packages API]
https://github.com/pulp/pulp_rpm/issues/3009[New setting for generated primary.xml timestamps to match upstream repository]
https://github.com/pulp/pulp_rpm/issues/2995[RPM replicator support]
https://github.com/pulp/pulpcore/issues/3538[New distribution hiding option]
https://github.com/pulp/pulpcore/issues/3835[OpenTelemetry support]
https://github.com/pulp/pulpcore/issues/3656[File size added to content application listing]
```

== Upgrade Warnings

[id="katello-deprecations"]

> == Removals

Reviewer: We always use the same headers (and remove them if they're empty). Please stick to them.

Author: @ekohl So here is the header incorrect? It should be Depreciations?

Reviewer: No, depreciation is a monetary thing. Quoting https://www.baeldung.com/cs/deprecated-vs-obsolete

> Depreciation is a term used to refer to the decrease in the value of an item or thing, such as a car. In programming, it is often used mistakenly to refer to deprecated.

Author: @jeremylenz recommended we label this as a Removal, in theforeman repo currently has Depreciations in it so I wasn't sure if that gets kept.
[foreman-documentation/blob/3.8/guides/doc-Release_Notes/topics/katello.adoc](https://github.com/theforeman/foreman-documentation/blob/3.8/guides/doc-Release_Notes/topics/katello.adoc?plain=1)

Author: I've pushed an update and removed empty headers

> === Katello-Agent has been removed.

Reviewer: I'd put this under Upgrade warnings.

And I'd also mention that it's automatically removed when upgrading, but that REX is not (automatically) installed as a replacement.

Author: committed!

Katello-Agent has been removed from Katello and is no longer available
via web UI, API, or CLI. For patching hosts going forward, 
use remote execution either in SSH or Pull mode.

> === Repositories now use pulp_deb 3.0.0.

Reviewer 3: I would combine this note with the one below and rename them to "Pulpcore and plugins upgraded to Pulpcore 3.28".

The note below this is actually a bugfix that happened mid-release, so it's technically not a change from 4.9 to 4.10

> Hammer: Allow user to run Verify Content Checksum, on container repositories.

Reviewer 3: Seems this line has a different format, as in it's missing the `===`

[id="katello-deprecations"]

> == Deprecations

Reviewer 5: ```suggestion
== Removals
```

Katello-agent removal is a removal, not a deprecation

> === Katello-Agent has been removed from Katello-host-tools; UI banners added to warn users.
> Removed API endpoints from Host, legacy ContentHost.

Reviewer 5: ```suggestion
=== Katello-Agent has been removed.
Katello-Agent has been removed from Katello and is no longer available via web UI, API, or CLI. For patching hosts going forward, use remote execution either in SSH or Pull mode.
```

=== Repositories now use pulp_deb 3.0.0.
Hammer: Allow user to run Verify Content Checksum, on container repositories.

=== Restore n-1 smart proxy sync support for pulpcore 3.28 / 3.22.
Fixed capsule sync issues and redundant sync in Foreman Proxy Content. 

=== Add ACS refresh option '--all' to "hammer alternate-content-source refresh" command, to refresh all ACS\'s.
ACS bulk refresh through API silently sanitizes input ids.

> === Remove the setting \'Use remote execution by default\'. 

Reviewer 5: ```suggestion
=== Remove the setting \'Use remote execution by default\'.
As part of the removal of katello-agent, the following related settings have been removed: \'Use remote execution by default\', \'Accept action timeout\', and \'Finish action timeout.\'
```

> === Add activation key details top bar.

Reviewer 5: ```suggestion
=== Add activation key details top bar.
When the setting 'Show Experimental Labs' is enabled, the Activation Key details page will be replaced by the beginnings of a redesigned page.
```

Author: committed this! see ln32 below
